app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -349,6 +357,62 @@ class _LinkRoomPageConsumerState extends ConsumerState<LinkRoomPage> {
       final room = await ref.watch(maybeRoomProvider(roomId).future);
       if (room != null) {
         room.addParentRoom(selectedParentSpaceId, true);
+        final joinRule = room.joinRuleStr();
+        List<String> currentRooms = [];
+        bool parentCanSee = joinRule == 'public';
+        String newRule = 'restricted';
+        if (joinRule == 'restricted' || joinRule == 'knock_restricted') {
+          currentRooms =
+              room.restrictedRoomIdsStr().map((t) => t.toString()).toList();
+          parentCanSee = currentRooms.contains(selectedParentSpaceId);
```

Comment:
Just to confirmed,

 when we have `restricted` or `knock_restricted` still it doesn't sufficient for space member can see it within space. 
 
 specific `room` should have `parentId` in `restrictedRoomIdsStr` to apply this join rule.
 
 `restricted` or `knock_restricted` without setting `parentId` in `restrictedRoomIdsStr` can be consider same as `invite` join_rule?

Reply:
That is correct. The "restricted" fields have a list roomIds (parents), who's member the access it restricted to. See https://spec.matrix.org/v1.8/client-server-api/#restricted-rooms